### PR TITLE
log-backup: add the switch for log backup

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/29501

Problem Summary:
we have several places need to check whether log backup is enabled.
1. Resolve locks to scan more locks after safepoint.
2. Add index to skip using lightning.
3. Telemetry of log backup feature usage statistics.

### What is changed and how it works?
1. Add a switch for log backup is enabled.
2. remove the old logic that check log backup config from http client.
3. This PR also fix two unstable test(owner_test, ddl_test) to make CI passed.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

when cluster not enable log backup or don't have the config `log-backup.enable`.
截屏2022-07-14 下午2 53 25

when enable to true.
image

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/backfilling.go [hunk 1]
```diff
@@ -609,6 +609,16 @@ func (w *worker) writePhysicalTableRecord(t table.PhysicalTable, bfWorkerType ba
 	}()
 	jc := w.jobContext(job)
 
+	// ctx, err := w.sessPool.get()
```
Comment (3pointer) [inline ddl/backfilling.go:612]: @Benjamin2037 Please help review this usage.

File: ddl/util/main_test.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ func TestMain(m *testing.M) {
 	testsetup.SetupForCommonTest()
 	opts := []goleak.Option{
 		goleak.IgnoreTopFunction("github.com/golang/glog.(*loggingT).flushDaemon"),
+		goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),
```
Comment (3pointer) [inline ddl/util/main_test.go:28]: @hawkingrei Please help review this fix of unit test.

File: telemetry/data_feature_usage.go [hunk 3]
```diff
@@ -47,6 +48,7 @@ type featureUsage struct {
 	NonTransactionalUsage *m.NonTransactionalStmtCounter   `json:"nonTransactional"`
 	GlobalKill            bool                             `json:"globalKill"`
 	MultiSchemaChange     *m.MultiSchemaChangeUsageCounter `json:"multiSchemaChange"`
+	LogBackup             bool                             `json:"logBackup"`
```
Comment (3pointer) [inline telemetry/data_feature_usage.go:51]: @AstroProfundis Please help review this change.

File: br/pkg/utils/db.go [hunk 4]
```diff
@@ -30,3 +35,49 @@ type DBExecutor interface {
...
 }
+
+// CheckLogBackupEnabled checks if LogBackup is enabled in cluster.
+// this mainly used in three places.
+// 1. Resolve locks to scan more locks after safepoint.
+// 2. Add index skipping use lightning.
+// 3. Telemetry of log backup feature usage statistics every 6 hours.
+// NOTE: this result shouldn't be cached by caller. because it may change every time in one cluster.
+func CheckLogBackupEnabled(ctx sessionctx.Context) bool {
+	enabled, err := IsLogBackupEnabled(ctx.(sqlexec.RestrictedSQLExecutor))
```
Comment (YuJuncen) [inline br/pkg/utils/db.go:46]: if we always assert the session context to this type, why not accept `sqlexec.RestrictedSQLExecutor` directly?

(Or we need to check whether the check successes, returning `true` when failed to make the assertion?)

File: br/pkg/utils/db.go [hunk 5]
```diff
@@ -30,3 +35,48 @@ type DBExecutor interface {
...
+// 3. Telemetry of log backup feature usage (every 6 hours).
+// NOTE: this result shouldn't be cached by caller. because it may change every time in one cluster.
+func CheckLogBackupEnabled(ctx sessionctx.Context) bool {
+	enabled, err := IsLogBackupEnabled(ctx.(sqlexec.RestrictedSQLExecutor))
+	if err != nil {
+		// if it failed by any reason. we can simply return true this time.
+		// for GC worker it will scan more locks in one tick.
+		// for Add index it will skip using lightning this time.
+		// for Telemetry it will get a false positive usage count.
+		return true
```
Comment (YuJuncen) [inline br/pkg/utils/db.go:52]: Maybe add some log here?

**Review by YuJuncen — APPROVED** [review]
rest lgtm

File: br/pkg/utils/db.go (hunk 4 above)
Comment (3pointer) [reply br/pkg/utils/db.go:46]: These three invoke can tolerate false positive when we get error(maybe internet error)

File: br/pkg/utils/db.go (hunk 5 above)
Comment (3pointer) [reply br/pkg/utils/db.go:54]: Done

File: br/pkg/utils/db.go (hunk 4 above)
Comment (YuJuncen) [reply br/pkg/utils/db.go:48]: Type assertion failure(without checking) can lead to panic, I'm not sure whether this is acceptable.

**Review by sticnarf — COMMENTED** [review]
The gc worker part lgtm.

**Review by AstroProfundis — APPROVED** [review]

**Pull request merged**